.github/workflows/azure-bicep-validate.yaml

```diff
@@ -11,11 +11,17 @@ on:
 jobs:
   build:
     runs-on: ubuntu-latest
+    permissions:
+      security-events: write
     steps:
       - name: Checkout
         uses: actions/checkout@v4
 
       - name: Azure CLI script
         uses: azure/CLI@v2
         with:
-          inlineScript: az config set bicep.use_binary_from_path=false && az bicep build -f infra/main.bicep
+          inlineScript: az bicep lint --file infra/main.bicep --diagnostics-format sarif > bicep-results.sarif
```

Comment:
@jamesc Do you know if this returns a failure exit code if the lint produces errors? I think bicep build does, but I cant tell if bicep lint does, based off the reference @ https://learn.microsoft.com/en-us/azure/azure-resource-manager/bicep/bicep-cli#lint

Reply:
Yep, looks like returns error codes correctly.  With only warnings:

``` bash
$ az bicep lint --file infra/main.bicep
/workspaces/openai-chat-app-entra-auth-builtin/infra/core/security/keyvault.bicep(6,7) : Warning no-unused-params: Parameter "principalId" is declared but never used. [https://aka.ms/bicep/linter/no-unused-params]
/workspaces/openai-chat-app-entra-auth-builtin/infra/core/host/container-app.bicep(36,7) : Warning no-unused-params: Parameter "postgresServiceId" is declared but never used. [https://aka.ms/bicep/linter/no-unused-params]
/workspaces/openai-chat-app-entra-auth-builtin/infra/main.bicep(19,3) : Warning no-conflicting-metadata: The "description" metadata property conflicts with the "description" decorator and will be overwritten. [https://aka.ms/bicep/linter/no-conflicting-metadata]
/workspaces/openai-chat-app-entra-auth-builtin/infra/core/ai/cognitiveservices.bicep(45,20) : Warning use-safe-access: Use the safe access (.?) operator instead of checking object contents with the 'contains' function. [https://aka.ms/bicep/linter/use-safe-access]
/workspaces/openai-chat-app-entra-auth-builtin/infra/core/ai/cognitiveservices.bicep(47,8) : Warning use-safe-access: Use the safe access (.?) operator instead of checking object contents with the 'contains' function. [https://aka.ms/bicep/linter/use-safe-access]
$ echo $?
0
```

With an error added:
``` bash
$ az bicep lint --file infra/main.bicep
/workspaces/openai-chat-app-entra-auth-builtin/infra/core/security/keyvault.bicep(6,7) : Warning no-unused-params: Parameter "principalId" is declared but never used. [https://aka.ms/bicep/linter/no-unused-params]
/workspaces/openai-chat-app-entra-auth-builtin/infra/core/host/container-app.bicep(36,7) : Warning no-unused-params: Parameter "postgresServiceId" is declared but never used. [https://aka.ms/bicep/linter/no-unused-params]
/workspaces/openai-chat-app-entra-auth-builtin/infra/main.bicep(19,3) : Warning no-conflicting-metadata: The "description" metadata property conflicts with the "description" decorator and will be overwritten. [https://aka.ms/bicep/linter/no-conflicting-metadata]
/workspaces/openai-chat-app-entra-auth-builtin/infra/core/ai/cognitiveservices.bicep(45,20) : Warning use-safe-access: Use the safe access (.?) operator instead of checking object contents with the 'contains' function. [https://aka.ms/bicep/linter/use-safe-access]
/workspaces/openai-chat-app-entra-auth-builtin/infra/core/ai/cognitiveservices.bicep(47,8) : Warning use-safe-access: Use the safe access (.?) operator instead of checking object contents with the 'contains' function. [https://aka.ms/bicep/linter/use-safe-access]
/workspaces/openai-chat-app-entra-auth-builtin/infra/main.bicep(18,13) : Error BCP071: Expected 1 argument, but got 2.
$ echo $?
1